storage/aptosdb/src/pruner/state_pruner_worker.rs

```diff
@@ -62,8 +62,9 @@ impl StatePrunerWorker {
     }
 
     pub fn set_target_db_version(&self, target_db_version: Version) {
-        assert!(target_db_version >= self.pruner.target_version());
```

Comment:
Why do we remove this assertion?

Reply:
The test I added in storage/aptosdb/src/aptosdb_test.rs, line 237 I "skip ahead" the pruner a bit to get a "clean DB" with all nodes referenced by some root, in order to test all that need to be pruned are marked by the stale node index. However that'll make this assertion fail.